src/marvin/core/ChatCompletion/providers/openai.py

```diff
@@ -204,6 +204,15 @@ async def _send_request_async(self, **serialized_request: Any) -> Response[T]:
         Send the serialized request to OpenAI's endpoint asynchronously.
         """
         import openai
+        
+        # Azure OpenAI Compatibility
+        api_type = serialized_request.get("api_type", None)
+        if isinstance(api_type, str) and api_type.startswith("azure"):
+            if deployment_name := serialized_request.pop("deployment_name", None):
+                serialized_request["deployment_id"] = deployment_name
+            else:
+                if model := serialized_request.pop("model", None):
+                    serialized_request["engine"] = model
```

Comment:
I think we should staff this in this area which is better to avoid distracting this function
```
marvin.core.ChatCompletion.providers.openai.OpenAIChatCompletion._serialize_request
```

Reply:
Hope this gets integrated soon!